middleware/throttle.go

```diff
@@ -2,23 +2,43 @@ package middleware
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/pressly/chi"
 	"golang.org/x/net/context"
 )
 
-// Throttle is a middleware that limits number of currently
-// processed requests at a time.
-func Throttle(limit int) func(chi.Handler) chi.Handler {
-	if limit <= 0 {
+var (
+	defaultTimeout = time.Second * 60
+)
+
+// Throttle is a middleware that limits number of currently processed requests
+// at a time.
+func Throttle(limit int, backloglimit int, timeout time.Duration) func(chi.Handler) chi.Handler {
```

Comment:
So far, it looks good! But before we merge this PR, I'd like to leave the original Throttle API backward-compatible. Backlog+timeout should be optional.

```go
middleware.Throttle(10) // hard limit on number of processed requests
middleware.Throttle(10, middleware.Backlog(1000, time.Second)) // use backlog+timeout too
```

@xiam @pkieltyka thoughts?